core/src/main/java/org/geysermc/geyser/skin/FakeHeadProvider.java

```diff
@@ -91,11 +94,38 @@ public SkinProvider.SkinData load(@Nonnull FakeHeadEntry fakeHeadEntry) throws E
                 }
             });
 
-    public static void setHead(GeyserSession session, PlayerEntity entity, CompoundTag profileTag) {
-        SkinManager.GameProfileData gameProfileData = SkinManager.GameProfileData.from(profileTag);
-        if (gameProfileData == null) {
-            return;
+    public static void setHead(GeyserSession session, PlayerEntity entity, CompoundTag NBTData) {
```

Comment:
Rename this variable to `nbtData` to comply a little better with variable naming schemes?